src/overlays/actors/ovl_En_Rd/z_en_rd.c

```diff
@@ -173,83 +205,97 @@ void EnRd_Destroy(Actor* thisx, GlobalContext* globalCtx) {
     if (gSaveContext.sunsSongState != SUNSSONG_INACTIVE) {
         gSaveContext.sunsSongState = SUNSSONG_INACTIVE;
     }
+
     Collider_DestroyCylinder(globalCtx, &this->collider);
 }
 
-void func_80AE2630(GlobalContext* globalCtx, Actor* thisx, s32 arg2) {
-    Actor* enemyIt = globalCtx->actorCtx.actorLists[ACTORCAT_ENEMY].head;
-
-    while (enemyIt != NULL) {
-        if ((enemyIt->id != ACTOR_EN_RD) || (enemyIt == thisx) || (enemyIt->params < 0)) {
-            enemyIt = enemyIt->next;
+/**
+ * This function does two things depending on whether shouldMourn is true or false.
+ * - If shouldMourn is true, this function sets thisx to be the parent for all Redeads
+ *   in the area that are capable of mourning. This is used right when thisx first
+ *   dies to make the other Redeads mourn it.
+ * - If shouldMourn is false, this function nulls out the parent for all Redeads in the
+ *   area whose parents are thisx. This is used when thisx is fading away to make the
+ *   other Redeads stop mourning over it.
+ */
```

Comment:
```
/**
 * The `parent` pointer if updated for all currently loaded redeads depending on the `shouldMourn` variable.
 *
 * If `shuoldMourn` is true, the parent of all other Redeads is set to this instance so they will mourn
 * this Redead.
 * 
 * If `shouldMourn` is false, the parent of all other Redeads is cleared so that they stop mourning.
 * This is done when the dead redead starts fading away
 */
 ```
Can add or tweak as needed, but imo I like to avoid references to actual variable names/code constructs in comments. At least when possible. So referring to "thisx" as "this instance" for example. The fact that its thisx is not important imo